CyberHealth/assessment/models.py

```diff
@@ -1,25 +1,74 @@
+from django.contrib.auth.models import User
 from django.db import models
 from django.utils.text import slugify
+from sort_order_field import SortOrderField
 
 
-class Question(models.Model):
-    question_text = models.CharField(max_length=255)
+class Control(models.Model):
+    title = models.CharField(max_length=255)
+    intro_text = models.CharField(max_length=255)
+    slug = models.SlugField(max_length=200, unique=True, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
 
     def __str__(self):
-        return self.question_text
+        return f"{self.id} - {self.title}"
+
+    def save(self, *args, **kwargs):
+        if self.slug:
+            # format slug if provided
+            self.slug = slugify(self.slug)
+        else:
+            # slugify title otherwise
+            self.slug = slugify(self.title)
+        
+        super(Control, self).save(*args, **kwargs)
 
 
-class Choice(models.Model):
-    choice_text = models.CharField(max_length=255)
-    question = models.ForeignKey(Question, on_delete=models.CASCADE)
+class SubControl(models.Model):
+    title = models.CharField(max_length=255)
+    sort_order = SortOrderField("Sort")
+    control = models.ForeignKey(Control, on_delete=models.CASCADE)
+    slug = models.SlugField(max_length=200, unique=True, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
 
     def __str__(self):
-        return self.choice_text
+        return f"{self.title} ({self.control})"
+
+    def save(self, *args, **kwargs):
+        if self.slug:
+            # format slug if provided
+            self.slug = slugify(self.slug)
+        else:
+            # slugify title otherwise
+            self.slug = slugify(self.title)
+        
+        super(SubControl, self).save(*args, **kwargs)
+
+    class Meta:
+        order_with_respect_to = 'control'
+        unique_together=['control', 'sort_order']
+
+
+class Question(models.Model):
+    question_text = models.CharField(max_length=255)
+    sub_control = models.ForeignKey(SubControl, on_delete=models.CASCADE, blank=True, null=True, related_name="in_sub_control")
+    created_at = models.DateTimeField(auto_now_add=True)
+    positive_answer = models.BooleanField(default=True)
+    sort_order = SortOrderField("Sort")
+    
+    def __str__(self):
+        return self.question_text
 
 
 class Answer(models.Model):
     question = models.ForeignKey(Question, on_delete=models.CASCADE)
-    choice = models.ForeignKey(Choice, on_delete=models.CASCADE)
+    response = models.BooleanField(
+            null=False,
+            choices=((True, 'Yes'), (False, 'No'))
+        )
+    created_at = models.DateTimeField(auto_now_add=True)
+    # TODO: change user to organisation_user when that data modelling has been done
```

Comment:
Can you add a task ticket in Jira rather than adding a TODO?

Reply:
An issue has been created! [Here it is](https://digital.dclg.gov.uk/jira/browse/LGCS-804) :~)